streams/src/main/java/org/apache/kafka/streams/state/internals/CachingSessionStore.java

```diff
@@ -46,7 +46,6 @@
     private final SessionKeySchema keySchema;
     private final SegmentedCacheFunction cacheFunction;
     private String cacheName;
-    private ThreadCache cache;
```

Comment:
nit: I feel it is not necessary to remove the extra reference on the cache and then call `context.cache()` every time we need it, but I think this is really a very very nit point so your call.

Reply:
Well, the underlying `ThreadCache` changes when we transition task type (as standbys have a dummy size-zero cache). So we need to go through the `context` to make sure we call the right cache